### PR TITLE
feat: upgrades react native template to 0.63.4

### PR DESCRIPTION
**Pull request by rynpatk**
**UPDATED 1/31/2021 to be 0.63.4 upgrade and also to replace splash-screen with bootsplash - cmejet**

This PR performs the necessary changes to upgrade the template from 0.62.2 to 0.63.2.

I have tested initializing new builds with this template locally and both iOS and Android are running.

### CODE REVIEW

File: template/ios/Podfile [hunk 1]
```diff
@@ -1,99 +1,33 @@
...
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
-  add_flipper_pods!
+  use_flipper!
   post_install do |installer|
     flipper_post_install(installer)
   end
 end
 
+target 'HelloWorld-tvOS' do
```
Comment (cmejet) [inline template/ios/Podfile:26]: They added tv OS back?? Do we really want these in our template?

**Comment by cmejet** [comment]
✅  Init with template (locally) was successful
✅  yarn test
✅  yarn ios
✅  yarn android
✅  yarn e2e:ios-debug
❌  yarn e2e:ios  (a flipper error) need to validate that a release build works
❌  yarn e2e:android-debug
❌  yarn e2e:android

File: template/ios/Podfile (hunk 1 above)
Comment (rynpatk) [reply template/ios/Podfile:26]: ahh I can probably remove this actually

**Comment by cmejet** [comment]
Updated configurations of Detox, getting closer!

✅  Init with template (locally) was successful
✅  yarn test
✅  yarn ios
✅  yarn android
✅  yarn e2e:ios-debug
❌  yarn e2e:ios (a flipper error) need to validate that a release build works
✅  yarn e2e:android-debug
✅  yarn e2e:android

File: template/src/App.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ class App extends Component<{}, State> {
       throw error;
     } finally {
       // Now that everything is hydrated, hide the splashscreen
-      SplashScreen.hide();
+      RNBootSplash.hide({ fade: true });
```
Comment (isaiahgrey93) [inline template/src/App.tsx:41]: Noice!

**Review by isaiahgrey93 — APPROVED** [review]

**Review by coryechobind — APPROVED** [review]

**Pull request merged**